src/main/java/com/redhat/devtools/lsp4ij/features/completion/LSPCompletionContributor.java

```diff
@@ -137,6 +136,51 @@ private void addCompletionItems(@NotNull CompletionParameters parameters,
             LookupElement lookupItem = completionFeature.createLookupElement(item, context);
             if (lookupItem != null) {
                 completionFeature.addLookupItem(context, completionPrefix, result, lookupItem, size - i, item);
+                ContainerUtil.addIfNotNull(addedLookupStrings, lookupItem.getLookupString());
+            }
+        }
+
+        // If completions were added from LSP and this is auto-completion or the explicit completion trigger was typed
+        // only once, add completions from other all other contributors except for the word completion contributor.
+        // Note that the word completion contributor only really kicks in when LSP completions are added in the context
+        // of a TextMate file (see TextMateCompletionContributor), but that's going to be very common for LSP usage
+        // since it's often adding (pseudo-)language support when not already present in the host JetBrains IDE.
+        if ((size > 0) && (parameters.getInvocationCount() < 2)) {
+            // Don't allow automatic execution of the remaining completion contributors; we'll execute them ourselves below
+            result.stopHere();
+
+            // Try to disable word completion by temporarily setting FORBID_WORD_COMPLETION=true on the completion process.
+            // Note that this will not work in some older IDE versions where WordCompletionContributor.addWordCompletionVariants()
```

Comment:
Added a comment about why this may not be working in older supported IDE versions in case that gets logged as an issue in the future.